### PR TITLE
v2 onboarding and PurgeCSS plugin documentation updates

### LINKED ISSUE #1849: v2 doc updates & PurgeCSS plugin resource
## v2 Doc Updates

The onboarding documentation has a few issues items that need to be adjusted:

- [x] The onboarding instructions for the Tailwind config should favor `.ts | .js | .cjs` in that tab order
- [x] The onboarding instructions for the Tailwind config need to recommend `.ts` for Typescript and `.js` for JS
- [x] When using Typescript, users will need to install default node types via: `npm add -D @types/node`

## Purge CSS plugin

Additionally, @AdrianGonz97 has given the thumbs up to document and point out his new PurgeCSS plugin for tree shaking Tailwind styles:

https://github.com/AdrianGonz97/vite-plugin-tailwind-purgecss

Let's go ahead and document this under Docs > Resources

### PR DESCRIPTION
**Pull request by endigo9740**
## Linked Issue

Closes #1849

## Description

Updates a number of issues within the onboarding documentation, as well as implementing the new PurgeCSS plugin documentation page.

## Changsets

Instructions: Changesets automate our changelog. If you modify files in `/packages/skeleton`, run `pnpm changeset` in the root of the monorepo, follow the prompts, then commit the markdown file. Changes that add features should be `minor` while chores and bugfixes should be `patch`. Please prefix the changeset message with `feat:`, `bugfix:` or `chore:`.

## Checklist

Please read and apply all [contribution requirements](https://www.skeleton.dev/docs/contributing).

- [x] This PR targets the `dev` branch (NEVER `master`)
- [x] Documentation reflects all relevant changes
- [x] Branch is prefixed with: `docs/`, `feat/`, `chore/`, `bugfix/`
- [x] Ensure Svelte and Typescript linting is current - run `pnpm check`
- [x] Ensure Prettier linting is current - run `pnpm format`
- [x] All test cases are passing - run `pnpm test`
- [x] Includes a changeset (if relevant; see above)

### CODE REVIEW

File: sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+<LayoutPage>
+	<!-- Header -->
+	<header class="space-y-4">
+		<h1 class="h1">PurgeCSS</h1>
+		<!-- prettier-ignore -->
+		<div>
+			<div class="!flex flex-col md:flex-row justify-between items-start space-y-4 md:space-y-0 md:space-x-4">
+                <!-- prettier-ignore -->
+				<p>
+                    This guide covers <code class="code">vite-plugin-tailwind-purgecss</code>, a simple to use Vite plugin that thoroughly purges excess CSS via <a class="anchor" href="https://purgecss.com/" target="_blank">PurgeCSS</a>. While optional, this is highly recommended if you wish to minimize your production CSS bundle size.
```
Comment (AdrianGonz97) [inline sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte:15]:
```suggestion
                    This guide covers <code class="code">vite-plugin-tailwind-purgecss</code>, a simple Vite plugin that thoroughly purges excess CSS via <a class="anchor" href="https://purgecss.com/" target="_blank">PurgeCSS</a>. While optional, this is highly recommended if you wish to minimize your production CSS bundle size.
```

File: sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+	<hr />
+
+	<section class="space-y-4">
+		<h2 class="h2">Introduction</h2>
+		<h3 class="h3">Motivation</h3>
+		<p>
+			Tailwind UI component libraries like Skeleton, Tailwind, and Flowbite provide a number of benefits, but come with an important caveat
+			- Tailwind generates classes for all imported components, regardless if they are used in the project or not. This leads to a larger
+			than necessary CSS bundle.
```
Comment (AdrianGonz97) [inline sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte:33]:
```suggestion
			Tailwind UI component libraries like Skeleton and Flowbite provide a number of benefits, but come with an important caveat
			- Tailwind generates classes for all imported components, regardless if they are used in the project or not. This leads to a larger
			than necessary CSS bundle.
```
Tailwind doesn't really fit in this list. This is mainly for component libraries that are built with Tailwind.

File: sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		</p>
+		<!-- prettier-ignore -->
+		<p>
+			Unfortunately this is a limitation of how Tailwind implements it's <a class="anchor" href="https://tailwindcss.com/docs/content-configuration" target="_blank">Content Configuration</a>. Tailwind searches through all files
+			specified in <code class="code">content</code>, uses a regex to locate possible selectors, then generates their respective classes. The key thing to note is this occurs before the build process, meaning there's no CSS treeshaking or purging involved against your production file assets.
+		</p>
+		<h3 class="h3">How it Works</h3>
+		<!-- prettier-ignore -->
+		<p>
+            Ideally, we would like to limit selectors to only those used within your project. We accomplish by analyzing the emitted Javascript chunks generated by Rollup, which includes only the modules that are actually used in your project, then extracts only the utilized selectors. From there, we can pass along the selectors to PurgeCSS for final processing.
```
Comment (AdrianGonz97) [inline sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte:43]:
```suggestion
            Ideally, we would like to limit selectors to only those used within your project. We accomplish this by analyzing the emitted Javascript chunks generated by Rollup, which includes only the modules that are actually used in your project, then it extracts the utilized selectors. From there, we can pass along the selectors to PurgeCSS for final processing.
```

File: sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+				greedy: [/^hljs-/],
+			},
+		}),
+	],
+};
+        `}
+		/>
+		<p>Alternatively, can safelist selectors directly within your stylesheets using special comments.</p>
+		<CodeBlock
+			language="ts"
```
Comment (AdrianGonz97) [inline sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte:94]:
```suggestion
			language="css"
```

File: sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+}\n
+h2 {
+	color: blue;
+}\n
+/*! purgecss end ignore */
+        `}
+		/>
+		<p>You may also safelist selectors directly within each declaration block.</p>
+		<CodeBlock
+			language="ts"
```
Comment (AdrianGonz97) [inline sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte:108]:
```suggestion
			language="css"
```

File: sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        </p>
+	</section>
+
+	<hr />
+
+	<section class="space-y-4">
+		<h2 class="h2">Attribution</h2>
+		<!-- prettier-ignore -->
+		<p>
+			This <a class="anchor" href="https://github.com/AdrianGonz97/vite-plugin-tailwind-purgecss" target="_blank" rel="noreferrer">plugin</a>  is provided courtesy of Skeleton co-maintainer <a class="anchor" href="https://github.com/joshnuss" target="_blank" rel="noreferrer">Adrian Gonz (aka CokaKoala)</a>.
```
Comment (AdrianGonz97) [inline sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte:128]:
```suggestion
			This <a class="anchor" href="https://github.com/AdrianGonz97/vite-plugin-tailwind-purgecss" target="_blank" rel="noreferrer">plugin</a>  is provided courtesy of Skeleton co-maintainer <a class="anchor" href="https://github.com/AdrianGonz97" target="_blank" rel="noreferrer">Adrian (aka CokaKoala)</a>.
```

File: sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte (hunk 2 above)
Comment (endigo9740) [reply sites/skeleton.dev/src/routes/(inner)/docs/purgecss/+page.svelte:33]: I meant to put Skeleton, Flowbite, and Daisy UI. I'll commit a fix shortly.

**Comment by endigo9740** [comment]
@Sarenor given the onboarding changes here I'll let you do a pass through my updates before we merge.

**Pull request merged**